app/models/event.rb

```diff
@@ -163,11 +171,31 @@ def previous_event
     Event.where('start_time < ?', self.start_time).order(start_time: :desc).first
   end
 
-private
+  def staffing_level
+    {
+      staffing_level_number: staffing_number,
+      staffing_level_name: STAFFING_LEVELS[staffing_number],
+      staffing_level_percentage: staffing_percentage
+    }
+  end
+
+  private
+
+  def staffing_number
+    tasks = self.tasks
+    return 0 if tasks.empty? # TODO: verify an event with no tasks should be considered empty.
+    @staffing_number ||= tasks.map(&:staffing_value).min
+  end
+
+  # TODO: update this once staffing % meaning is clarified.
+  def staffing_percentage
+    staffing_number * 20 + 20
+  end
+
   def calc_duration #This is also used in timecards; it should be extracted out
-     if !(start_time.blank?) and !(end_time.blank?)
+    if !(start_time.blank?) and !(end_time.blank?)
```

Comment:
The `and` here might give us unintended consequences because precedence. 
Additionally, it's better to use `if start_time.present? && end_time.present?`